lib/matplotlib/axes/_secondary_axes.py

```diff
@@ -0,0 +1,581 @@
+            q = self._forward(values)
+        return q
+
+    def inverted(self):
+        return _ArbitraryTransform(self._inverse, self._forward)
+
+
+class ArbitraryScale(mscale.ScaleBase):
+
+    name = 'arbitrary'
+
+    def __init__(self, axis, transform=mtransforms.IdentityTransform()):
+        """
+        TODO
+        """
+        self._transform = transform
+
+    def get_transform(self):
+        """
+        The transform for linear scaling is just the
+        :class:`~matplotlib.transforms.IdentityTransform`.
+        """
+        return self._transform
+
+    def set_default_locators_and_formatters(self, axis):
+        """
+        Set the locators and formatters to reasonable defaults for
+        linear scaling.
+        """
+        axis.set_major_locator(AutoLocator())
+        axis.set_major_formatter(ScalarFormatter())
+        axis.set_minor_formatter(NullFormatter())
+
+mscale.register_scale(ArbitraryScale)
+
+
+class ArbitraryLogScale(mscale.ScaleBase):
+
+    name = 'arbitrarylog'
+
+    def __init__(self, axis, transform=mtransforms.IdentityTransform()):
+        """
+        TODO
+        """
+        self._transform = transform
+
+    def get_transform(self):
+        """
+        The transform for linear scaling is just the
+        :class:`~matplotlib.transforms.IdentityTransform`.
+        """
+        return self._transform + Log10Transform()
+
+    def set_default_locators_and_formatters(self, axis):
+        """
+        Set the locators and formatters to reasonable defaults for
+        linear scaling.
+        """
+        self.base = 10
+        self.subs = None
+        axis.set_major_locator(LogLocator(self.base))
+        axis.set_major_formatter(LogFormatterSciNotation(self.base))
+        axis.set_minor_locator(LogLocator(self.base, self.subs))
+        axis.set_minor_formatter(
+            LogFormatterSciNotation(self.base,
+                                    labelOnlyBase=(self.subs is not None)))
+
+mscale.register_scale(ArbitraryLogScale)
+
+
+_secax_docstring ='''
+Warnings
+--------
+
+This method is experimental as of 3.1, and the API may change.
+
+Parameters
+----------
+location : string or scalar
+    The position to put the secondary axis.  Strings can be 'top' or
+    'bottom', scalar can be a float indicating the relative position
+    on the axes to put the new axes (0 being the bottom, and 1.0 being
+    the top.)
+
+conversion : scalar, 2-tuple of scalars, 2-tuple of func, or Transform
+    If a scalar or a 2-tuple of scalar, the secondary axis is
+    converted via a linear conversion with slope given by the first
+    and offset given by the second.  i.e. ``conversion = [2, 1]``
+    element for a parent axis between 0 and 1 gives a secondary axis
+    between 1 and 3.
+
+    If a 2-tuple of functions, the user specifies the transform
+    function and its inverse.  i.e.
+    `conversion=(lamda x: 2 / x, lambda x: 1 / (2 * x))` would be an
```

Comment:
"lambda" with a *b*. 

The inverse of `lambda x: 2 / x` should be `lambda x: 2 / x`, i.e. the function itself, right?